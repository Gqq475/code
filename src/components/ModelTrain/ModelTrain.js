import React from 'react'
import classes from './ModelTrain.scss'
import ModelTrainList from 'components/ModelTrainList'
import ModelTrainDetail from 'components/ModelTrainDetail'
import CreateModelDialog from 'components/CreateModelDialog'
import { assign, findIndex } from 'lodash'

type Props = {
  data: Object,
  createTrainModel: Function,
  getModelInfo: Function,
  getModelTrainResult: Function,
  updateModelTrainList: Function,
  getModelTrainAll: Function,
  getTestResultType: Function,
  getModelTestCount: Function,
  getModelResultList: Function,
  getModelTestAll: Function
};

export class ModelTrain extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.state = {
      detailData: {},
      isOpen: false,
      isBeginTraning: false,
      trainingDetail: false,
      testingDetail: false,
      trainStatus: ''
    }
    this.setDetail = this.setDetail.bind(this)
    this.toggleDialog = this.toggleDialog.bind(this)
    this.changeState = this.changeState.bind(this)
  }

  toggleDialog () {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  setDetail (data, trainingDetail, testingDetail) {
    this.setState(assign({}, this.state, {
      detailData: data,
      isBeginTraning: false,
      trainingDetail: trainingDetail,
      testingDetail: testingDetail
    }))
  }
  changeState (trainingDetail, testingDetail) {
    this.setState({
      trainingDetail: trainingDetail,
      testingDetail: testingDetail
    })
  }

  componentWillReceiveProps (nextProps) {
    if (this.state.detailData.id) {
      let index = findIndex(nextProps.data.modelTrainList, ['id', this.state.detailData.id])
      this.setState(assign({}, this.state, {
        detailData: nextProps.data.modelTrainList[index]
      }))
    }
  }

  render () {
    return (
      <div className={classes['ModelTrain-container']}>
        <section className={`${classes.title} ${classes.clear}`}>
          <h4 className={`${classes.left}`}>模型列表
            <button type='button' onClick={this.toggleDialog}
              className={`${classes.ml10} pt-button pt-intent-primary pt-icon-plus`} />
            {
              this.state.isOpen
              ? <CreateModelDialog
                isOpen={this.state.isOpen}
                toggleDialog={this.toggleDialog}
                createTrainModel={this.props.createTrainModel}
                />
              : null
            }
          </h4>
        </section>
        <ModelTrainList
          getModelTrainAll={this.props.getModelTrainAll}
          trainingDetail={this.state.trainingDetail}
          testingDetail={this.state.testingDetail}
          setDetail={this.setDetail} />
        {
          <ModelTrainDetail
            trainingDetail={this.state.trainingDetail}
            testingDetail={this.state.testingDetail}
            changeState={this.changeState}
            getModelTestAll={this.props.getModelTestAll}
            updateModelTrainList={this.props.updateModelTrainList}
            getModelTrainAll={this.props.getModelTrainAll}
            getModelInfo={this.props.getModelInfo}
            getModelTrainResult={this.props.getModelTrainResult}
            getModelResultList={this.props.getModelResultList}
            getTestResultType={this.props.getTestResultType}
            getModelTestCount={this.props.getModelTestCount}
            data={this.props.data}
            detailData={this.state.detailData} />
        }
      </div>
    )
  }
}

export default ModelTrain
