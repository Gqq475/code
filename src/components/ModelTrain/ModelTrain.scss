.ModelTrain-container {
  .clear{

    &:after{
      content: '';
      display: block;
      clear: both;
    }
  }
  .right{
    float: right;
  }
  .left{
    float: left;
  }
  .mr10{
    margin-right: 10px;
  }
  .ml10{
    margin-left: 10px;
  }
  .c1{
    color: #0087DC;
  }
  .cn{
    text-align: center;
  }

  .action{
    position: absolute;
    top: 30px;
    right: 140px;
  }

  .eye{
    position: absolute;
    top: 30px;
    right: 82px;
  }

  .log{
    position: absolute;
    top: 30px;
    right: 20px;
  }

  .title{
    background: #F7F7F7;
    padding: 10px;
    position: relative;
  }
}
