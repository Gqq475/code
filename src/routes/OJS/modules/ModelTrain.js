import {
  GET_MODEL_TRAIN_LIST,
  SUCCESSFULLY_CREATE_TRAINING_MODEL,
  GET_MODEL_TRAIN_INFO,
  GET_MODEL_TRAIN_RESULT_CHART,
  SUCCESSFULLY_UPDATE_TRAINING_MODEL,
  GET_MODEL_TRAIN_ALL,
  FAILED_CREATE_TRAINING_MODEL
} from './OJS'
import OurToaster from 'components/OurToaster'

export const getModelTrainList = (date) => (dispatch, getState) => {
  fetch(`${__QCA__}/offline/version_control/model_versions`, {
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then(json => {
    if (json.code === 200000) {
      dispatch({
        type: GET_MODEL_TRAIN_LIST,
        data: json.result.models
      })
    }
  })
  .catch(err => console.log(err))
}

export const createTrainModel = (state) => (dispatch, getState) => {
  fetch(`${__QCA__}/aiModel/model`, {
    method: 'POST',
    body: JSON.stringify({
      'path': 'E:/Model/Save/model',
      'name': state.name,
      'imageResource': state.imageResource,
      'lossRate': state.lossRate,
      'iteration': state.iteration,
      'gpuId': state.gpuId
    }),
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then(data => {
    if (data.code === 200000) {
      dispatch({
        type: SUCCESSFULLY_CREATE_TRAINING_MODEL,
        // work around
        data: data.result
      })
      OurToaster.show({message: '添加成功！'})
    } else {
      dispatch({
        type: FAILED_CREATE_TRAINING_MODEL,
        // work around
        data: 'failed'
      })
      OurToaster.show({message: '添加失败！'})
    }
  })
  .catch(err => {
    console.log(err)
    OurToaster.show({message: '添加失败！'})
  })
}

export const getModelInfo = (id) => (dispatch, getState) => {
  fetch(`${__QCA__}/aiModel/modelInfo?id=${id}`, {
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then(json => {
    if (json.code === 200000) {
      dispatch({
        type: GET_MODEL_TRAIN_INFO,
        data: json.result
      })
    }
  })
  .catch(err => console.log(err))
}

export const getModelTrainResult = (id) => (dispatch, getState) => {
  fetch(`${__QCA__}/aiModel/imageTypeCount?id=${id}`, {
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: GET_MODEL_TRAIN_RESULT_CHART,
        data: data.result
      })
    }
  })
  .catch(err => console.log(err))
}

export const updateModelTrainList = (id, state) => (dispatch, getState) => {
  fetch(`${__QCA__}/aiModel/model`, {
    method: 'PUT',
    body: JSON.stringify({
      'id': id,
      'productCode': state.product,
      'operationCode': state.operation,
      'eqCode': state.subEntity,
      'lineCode': state.lineID
    }),
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then(data => {
    if (data.code === 200000) {
      OurToaster.show({message: '修改成功！'})
      dispatch({
        type: SUCCESSFULLY_UPDATE_TRAINING_MODEL,
        payload: {
          'id': id,
          'productCode': state.product,
          'operationCode': state.operation,
          'eqCode': state.subEntity,
          'lineCode': state.lineID
        }
      })
    } else {
      OurToaster.show({message: '修改失败'})
    }
  })
  .catch(err => {
    console.log(err)
    OurToaster.show({message: '修改失败'})
  })
}

export const getModelTrainAll = (id) => (dispatch, getState) => {
  fetch(`${__QCA__}/aiModel/trainProgress?id=${id}`, {
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
  .then(response => response.json())
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: GET_MODEL_TRAIN_ALL,
        data: data.result
      })
    }
  })
  .catch(err => console.log(err))
}
