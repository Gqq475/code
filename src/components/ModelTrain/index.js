import ModelTrain from './ModelTrain'
export default ModelTrain
