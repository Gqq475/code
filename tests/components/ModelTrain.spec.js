import React from 'react'
import { ModelTrain } from 'components/ModelTrain/ModelTrain'

describe('(Component) ModelTrain', () => {
  it('render normally', () => {
    const data = {
      'modelTrainList': [
        {
          defectTypeCount: 1,
          eqCode: 'TestEqCode',
          id: 317
        },
        {
          defectTypeCount: 5,
          eqCode: 'TestEqCode',
          id: 3
        }
      ]
    }
    const wrapper = shallow(
      <ModelTrain data={data} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
