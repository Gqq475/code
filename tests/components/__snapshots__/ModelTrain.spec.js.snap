// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelTrain render normally 1`] = `
<div
  className="ModelTrain-container"
>
  <section
    className="title clear"
  >
    <h4
      className="left"
    >
      模型列表
      <button
        className="ml10 pt-button pt-intent-primary pt-icon-plus"
        onClick={[Function]}
        type="button"
      />
    </h4>
  </section>
  <Connect(ModelTrainList)
    setDetail={[Function]}
    testingDetail={false}
    trainingDetail={false}
  />
  <Connect(ModelTrainDetail)
    changeState={[Function]}
    data={
      Object {
        "modelTrainList": Array [
          Object {
            "defectTypeCount": 1,
            "eqCode": "TestEqCode",
            "id": 317,
          },
          Object {
            "defectTypeCount": 5,
            "eqCode": "TestEqCode",
            "id": 3,
          },
        ],
      }
    }
    detailData={Object {}}
    testingDetail={false}
    trainingDetail={false}
  />
</div>
`;
